dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/StateWheelExecuteThread.java

```diff
@@ -84,16 +85,17 @@ public class StateWheelExecuteThread extends Thread {
 
     @Override
     public void run() {
+        Duration checkInterval = Duration.ofMillis(masterConfig.getStateWheelInterval() * Constants.SLEEP_TIME_MILLIS);
         while (Stopper.isRunning()) {
             try {
                 checkTask4Timeout();
                 checkTask4Retry();
                 checkTask4State();
                 checkProcess4Timeout();
+                ThreadUtils.sleep(checkInterval);
```

Comment:
I think it may cause rerun instantly without sleep if there are some exceptions before this...

Reply:
Good catch.